Sources/ManageSmileys.php

```diff
@@ -702,14 +702,15 @@ function AddSmiley()
 					fatal_lang_error('smileys_upload_error_blank');
 
 				if (empty($newName))
-					$newName = basename($_FILES[$name]['name']);
-				elseif (basename($_FILES[$name]['name']) != $newName)
+					$newName = pathinfo($_FILES[$name]['name'], PATHINFO_FILENAME);
+				elseif (pathinfo($_FILES[$name]['name'], PATHINFO_FILENAME) != $newName)
 					fatal_lang_error('smileys_upload_error_name');
 			}
 
 			foreach ($context['smiley_sets'] as $i => $set)
 			{
-				$set['name'] = un_htmlspecialchars($set['name']);
+				// Name must be treated differently; it's used within a variable name & php converted spaces & periods to underscores
+				$set['name'] = strtr(un_htmlspecialchars($set['name']), ' .', '__');
```

Comment:
I'm not sure why this change is necessary. A variable name (e.g. `$foo`) can't contain spaces or dots, true, but the index keys of an array certainly can. For example, this is perfectly valid: `$foo['I am the very model of a modern Major-General'] = "I've information vegetable, animal, and mineral";`. Since the names of the individual smiley files are not written into our code as variables, I don't think you need to do this.

Also, if you do this, how will you handle converting these back to the actual names when required?